sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -270,6 +270,8 @@ private[sql] object PartitioningUtils {
   private val upCastingOrder: Seq[DataType] =
     Seq(NullType, IntegerType, LongType, FloatType, DoubleType, StringType)
 
+  val validPartitionColumnTypes: Set[DataType] = upCastingOrder.toSet
```

Comment:
Which types are considered valid type for partition columns? The types have well defined string representation? If so I think I should add ByteType, ShortType, DecimalType, DateType and TimestampType here.